include/sys/sdt.h

```diff
@@ -55,15 +51,9 @@
  * twice, so it should not have side effects (e.g. something like:
  * "return (SET_ERROR(log_error(EINVAL, info)));" would log the error twice).
  */
-#define	SET_ERROR(err) \
-	(trace_zfs_set__error(__FILE__, __func__, __LINE__, err), err)
-
-#else
-
 #undef SET_ERROR
-#define	SET_ERROR(err) (err)
-
-#endif /* HAVE_DECLARE_EVENT_CLASS */
+#define	SET_ERROR(err) \
+	(__dprintf(__FILE__, __func__, __LINE__, "error %lu", err), err)
```

Comment:
We should be able to simplify this further by defining `SET_ERROR` to use `__dprintf` in both user and kernel space.

```diff
diff --git a/include/sys/sdt.h b/include/sys/sdt.h
index 3503f24..c76c2f0 100644
--- a/include/sys/sdt.h
+++ b/include/sys/sdt.h
@@ -34,9 +34,8 @@
 #define        ZFS_PROBE2(a, c, e)             ((void) 0)
 #define        ZFS_PROBE3(a, c, e, g)          ((void) 0)
 #define        ZFS_PROBE4(a, c, e, g, i)       ((void) 0)
-#define        ZFS_SET_ERROR(err)              ((void) 0)
 
-#else
+#endif /* _KERNEL */
 
 /*
  * The set-error SDT probe is extra static, in that we declare its fake
@@ -55,6 +54,5 @@
 #define        SET_ERROR(err) \
        (__dprintf(__FILE__, __func__, __LINE__, "error %lu", err), err)
 
-#endif /* _KERNEL */
 
 #endif /* _SYS_SDT_H */
diff --git a/include/sys/zfs_context.h b/include/sys/zfs_context.h
index 6163b2f..4fe3534 100644
--- a/include/sys/zfs_context.h
+++ b/include/sys/zfs_context.h
@@ -207,15 +207,6 @@ extern int aok;
        (unsigned long)i)
 
 /*
- * We use the comma operator so that this macro can be used without much
- * additional code.  For example, "return (EINVAL);" becomes
- * "return (SET_ERROR(EINVAL));".  Note that the argument will be evaluated
- * twice, so it should not have side effects (e.g. something like:
- * "return (SET_ERROR(log_error(EINVAL, info)));" would log the error twice).
- */
-#define        SET_ERROR(err) (ZFS_SET_ERROR(err), err)
-
-/*
  * Threads.  TS_STACK_MIN is dictated by the minimum allowed pthread stack
  * size.  While TS_STACK_MAX is somewhat arbitrary, it was selected to be
  * large enough for the expected stack depth while small enough to avoid
diff --git a/lib/libzfs/libzfs_dataset.c b/lib/libzfs/libzfs_dataset.c
```